src/integrations/prefect-redis/tests/test_locking.py

```diff
@@ -38,19 +38,26 @@ def test_init(self):
     def test_read_locked_key(self, store):
         key = str(uuid4())
         read_queue = queue.Queue()
+        thread_started = threading.Event()
 
         def read_locked_key():
+            thread_started.set()
+
+            # This read should block until the lock is released
             record = store.read(key)
             assert record is not None
             read_queue.put(record.result, block=False)
 
-        thread = threading.Thread(
-            target=read_locked_key,
-        )
+        thread = threading.Thread(target=read_locked_key)
+
         assert store.acquire_lock(key, holder="holder1")
+
         thread.start()
+        thread_started.wait()
+
         store.write(key=key, obj={"test": "value"}, holder="holder1")
         store.release_lock(key, holder="holder1")
+
         # the read should have been blocked until the lock was released
```

Comment:
It's really weird that I needed to update these tests, but I think we might have just had a latent flake that was surfaced by adding new tests that changed the run order.  This test started failing deterministically, only on Python 3.9, only in CI :facepalm:  It took some experimenting but adding this event to signal when the thread actually started seemed to do the trick.